### PR TITLE
NEP29: Set minimum required version to Python 3.8+

### PR DESCRIPTION
**Pull request by michaelgrund**
**Description of proposed changes**

- Remove Python 3.7 from our CI jobs
- Bump the minimum required Python version to 3.8 in the documentation
- Remove Python 3.7 from setup.py

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: .github/workflows/ci_tests.yaml [hunk 1]
```diff
@@ -30,7 +30,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: [3.7, 3.9]
+        python-version: [3.8, 3.9]
```
Comment (weiji14) [inline .github/workflows/ci_tests.yaml:33]:
```suggestion
        python-version: ['3.8', '3.9']
```

File: .github/workflows/ci_tests.yaml [hunk 2]
```diff
@@ -44,10 +44,10 @@ jobs:
           # - os: ubuntu-latest
           #   python-version: 3.7
           #   isDraft: true
-        # Pair Python 3.7 with NumPy 1.18 and Python 3.9 with NumPy 1.21
+        # Pair Python 3.8 with NumPy 1.19 and Python 3.9 with NumPy 1.21
         # Only install optional packages on Python 3.9/NumPy 1.21
         include:
-          - python-version: 3.7
+          - python-version: 3.8
```
Comment (weiji14) [inline .github/workflows/ci_tests.yaml:50]:
```suggestion
          - python-version: '3.8'
```

File: .github/workflows/ci_tests.yaml [hunk 3]
```diff
@@ -44,10 +44,10 @@ jobs:
...
             numpy-version: '1.18'
             optional-packages: ''
           - python-version: 3.9
```
Comment (weiji14) [inline .github/workflows/ci_tests.yaml:53]:
```suggestion
          - python-version: '3.9'
```

File: .github/workflows/ci_tests.yaml [hunk 4]
```diff
@@ -44,10 +44,10 @@ jobs:
           # - os: ubuntu-latest
           #   python-version: 3.7
           #   isDraft: true
-        # Pair Python 3.7 with NumPy 1.18 and Python 3.9 with NumPy 1.21
+        # Pair Python 3.8 with NumPy 1.19 and Python 3.9 with NumPy 1.21
```
Comment (weiji14) [inline .github/workflows/ci_tests.yaml:47]: Need to merge #1675 first, otherwise will conflict with this change here.

File: setup.py [hunk 5]
```diff
@@ -27,7 +27,6 @@
     "Intended Audience :: Education",
     "Topic :: Scientific/Engineering",
     "Topic :: Software Development :: Libraries",
-    "Programming Language :: Python :: 3.7",
```
Comment (weiji14) [inline setup.py:30]: Could you also change the L36 below to read: `PYTHON_REQUIRES = ">=3.8"`? Thanks!

**Review by weiji14 — COMMENTED** [review]
Thanks @michaelgrund! Just some suggestions to put quotes around the version numbers (otherwise Python 3.10 will be read as Python 3.1 in the future, see #1577). I'll reduce the diffs in #1577 a little so that it won't conflict with this PR (edit: done at fca7a689947191ffa694d184901a8f4e9559e47c).

**Review by weiji14 — APPROVED** [review]
Thanks @michaelgrund, I've changed the PR's title and updated the required checks at https://github.com/GenericMappingTools/pygmt/settings. Just wait for a second approval before merging.

**Review by seisman — APPROVED** [review]

**Pull request merged**